File: 3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.java
class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {
        int total1 = 0, count = 0,total2 = 0;
        for(int i : apple) total1 += i;
        Arrays.sort(capacity);
        for(int i = capacity.length - 1; i >= 0;i--){
            total2 += capacity[i];
            count++;
            if(total2 >= total1) break;
        }
        return count;
    }
}

File: 3334-apple-redistribution-into-boxes/apple-redistribution-into-boxes.rs
impl Solution {
    pub fn minimum_boxes(apple: Vec<i32>, mut capacity: Vec<i32>) -> i32 {
        let total1: i32 = apple.iter().sum();
        let mut count: i32 = 0;
        let mut total2: i32 = 0;
        capacity.sort_by(|a, b| b.cmp(a));
        for cap in capacity {
            total2 += cap;
            count += 1;
            if total2 >= total1 {
                break;
            }
        }
        count
    }
}
